sympy/stats/random_matrix_models.py

```diff
@@ -212,6 +214,45 @@ def joint_eigen_distribution(mat):
     """
     return mat.pspace.model.joint_eigen_distribution()
 
+@dispatch(Matrix)
+def joint_eigen_distribution(mat):
```

Comment:
Should this be named `JointEigenDistribution` (even though it is not a class, it still returns an unevaluated object)?